### DISCUSSION TITLE
Suggestion: Global Variables

### DISCUSSION THREAD

**Discussion by raharth** [FreeCAD/FreeCAD#10825] [Discussions that will turn in to issues but need refining]
I know, I know it's always the new guys that come up with stupid ideas, but I was not able to find anything on the topic in the tickets or previous discussions.

I think it would be really handy if there is a set of global variables that one can refer to. Right now when building my sketches I always have to go through the list of parts identifying the one in which I defined the contraint. At least for me it would be tremendously helpful if one could either set global variables as some kind of list or have some list of defined variables independent of in which sketch they were defined. Currently I build that list in a notepad next to FreeCAD so that I can simply copy them, but having them in the tool itself would be great I think.

**Comment by chennes** [comment]
Have you considered using the Spreadsheet workbench? You can give a cell an "alias" that you can then refer to throughout the program.

**Reply by raharth** [reply]
My bad, that looks quite similar to what I thought of! Thank you, I had totally missed that!
